File: 5.longest-palindromic-substring/solution.java
class Solution {
    public String longestPalindrome(String s) {
        StringBuffer sb = new StringBuffer("#");
		int len = s.length();
		for (int i = 0; i < len; i++)
			sb.append(s.charAt(i)).append('#');
		s = sb.toString();
		int maxLen = 1;
		len = s.length();
		int maxCenter = 1;
		for (int center = 2; center + maxLen < len; center++)
		{
			int j = 1;
			for (; center - j >= 0 && center + j < len; j++)
				if (s.charAt(center - j) != s.charAt(center + j))
					break;
			j--;
			if (j > maxLen)
			{
				maxLen = j;
				maxCenter = center;
			}
		}
		String maxAns = s.substring(maxCenter - maxLen, maxCenter + maxLen + 1);
		len = maxAns.length() / 2;
		sb = new StringBuffer();
		for (int i = 0; i < len; i++)
			sb.append(maxAns.charAt(i * 2 + 1));
		return sb.toString();
    }
}

File: 5.longest-palindromic-substring/solution.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        ss = "#" + "#".join(list(s)) + "#"
        max_len = 0
        val = ""
        for i in range(len(ss)):
            j = 1
            while i - j >= 0 and i + j < len(ss) and ss[i - j] == ss[i + j]:
                j += 1
            if max_len < j - 1:
                max_len = j - 1
                val = ss[i - j + 1:i + j]
        return val.replace("#", "")
